credsweeper/credentials/augment_candidates.py

```diff
@@ -0,0 +1,16 @@
+from typing import List
+
+from credsweeper.credentials import Candidate
+
+
+def augment_candidates(candidates: List[Candidate], new_candidates: List[Candidate]):
+    """Augment candidates with new_candidates if value of line data is not present in the candidates"""
+    if not new_candidates:
+        return
+    found_values = set(line_data.value for candidate in candidates  #
+                       for line_data in candidate.line_data_list)
+    for new_candidate in new_candidates:
+        for line_data in new_candidate.line_data_list:
+            if line_data.value not in found_values:
+                candidates.append(new_candidate)
+                break
```

Comment:
I think it would be good to add `return` line in `augment_candidates()`.
For good readability like below.

``` python
...

candidates.extend(augment_candidates(new_candidates))

...
```

Reply:
``augment_candidates`` requires a list which has to be augmented with new candidates.
If a new candidate has the same value - it means "upper" layer of scanner (plain text usually) already found the credential.
The function reduces multiple duplicates of credential. E.g. GitHub token may be found in text which is a YAML or Json. Then, deep scan finds the same value when markup language is parsed to structure.
Only the value was chosen as key due line number and file name might be not defined in the scan.

Reply:
Yup, I know about the method.
I just worried about the style of method and I think it can be reduce readability which method without return value.
Anyway if you think it is good, its okay for me too.

Reply:
@csh519 
docsting was updated

Reply:
Thank you for update.